packages/pnpm-sync/src/start.ts

```diff
@@ -58,7 +65,23 @@ program
           if (lockfile === null) {
             return undefined;
           } else {
-            return lockfile;
+            const lockfilePackages: Record<string, ILockfilePackage> = {};
```

Comment:
This is not a safe way to create a `Record<>`.  For example:

```ts
const lockfilePackages: Record<string, ILockfilePackage> = {};

function oops(packagePath: string): boolean {
  if (lockfilePackages[packagePath]) {
    console('The key is present')
  }
}

// These will be "found" in the empty object
oops('constructor')
oops('__proto__')
```

To avoid this you could use:
```ts
const lockfilePackages: Record<string, ILockfilePackage> = Object.create(null);
```
which is what the YAML parser does.